src/ContainerCheckpointModal.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Modal } from 'patternfly-react';
+import { Button, Checkbox } from '@patternfly/react-core';
+import cockpit from 'cockpit';
+import * as utils from './util.js';
+
+const _ = cockpit.gettext;
+
+class ContainerCheckpointModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            keep: false,
+            leaveRunning: false,
+            tcpEstablished: false,
+            ignoreRootFS: false
+        };
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(checked, event) {
+        if (event.target.type === "checkbox")
+            this.setState({ [event.target.name]: event.target.checked });
+    }
+
+    render() {
+        return (
+            <Modal show={this.props.selectContainerCheckpointModal}>
+                <Modal.Header>
+                    <Modal.Title>
+                        {cockpit.format(_("Checkpoint container $0"),
+                                        utils.truncate_id(this.props.containerWillCheckpoint.Id))}
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <div className="ct-form">
+                        <Checkbox label={_("Keep all temporary checkpoint files")} id="keep" name="keep"
+                                  isChecked={this.state.keep} onChange={this.handleChange} />
+                        <Checkbox label={_("Leave the container running after writing checkpoint to disk")}
+                                  id="leaveRunning" name="leaveRunning" isChecked={this.state.leaveRunning}
+                                  onChange={this.handleChange} />
+                        <Checkbox label={_("Checkpoint a container with established TCP connections")}
```

Comment:
Is this the container you're working with or another container? For me it reads as the latter.
Would need to rephrase for clarity. Would `remember established TCP connections` work? (if that is what it does, I'm not super-familiar with exactly what this option does)

Reply:
Good point - I copied the description of the options from the API documentation and from Podman command line help, but the context is clearly different here. I don't think that "remember" is the right word for what the option does, `support preserving established TCP connections` sounds better to me.

Reply:
A similar problem is in the restore option description, where simply dropping the `a container` (making it `restore with established TCP connections`) should suffice.

Reply:
Sounds good to me!